for_012_fire_risk/contents/src/__init__.py

```diff
@@ -11,267 +11,466 @@
     else:
         for i in range(10):
-            datestr = date.strftime(DATE_FORMAT_NETCDF)
-            new_dates.append(datestr)  #add to new dates
-            date -= datetime.timedelta(**TIMESTEP)
+            # generate a string from the date
+            datestr = date.strftime(DATE_FORMAT)
+            # add to list of new dates
+            new_dates.append(datestr) 
+            # go back one day at a time
+            date -= datetime.timedelta(days=1)
     return new_dates
 
 def convert(files):
-    '''convert netcdfs to tifs'''
+    '''
+    Convert netcdf files to tifs
+    INPUT   files: list of file names for netcdfs that have already been downloaded (list of strings)
+    RETURN  tifs: list of file names for tifs that have been generated (list of strings)
+    '''
+    # create an empty list to store the names of the tifs we generate from all input netcdf files
     tifs = []
+    # go through each netcdf file and translate
     for f in files:
+        # create an empty list to store the names of the tifs we generate from this netcdf file
         band_tifs = []
+        # go through each variables to process in this netcdf file
         for sds_name in SDS_NAMES:
             # extract subdataset by name
             sds_path = sds_name.format(fname=f)
-            band_tif = '{}_{}.tif'.format(os.path.splitext(f)[0], sds_name.split('_')[-1]) #naming tiffs
-            cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, band_tif] #'-q' means quiet so you don't see it
+            # generate a name to save the tif file we will translate the netcdf file's subdataset into
+            band_tif = '{}_{}.tif'.format(os.path.splitext(f)[0], sds_name.split('_')[-1]) 
+            # translate the netcdf file's subdataset into a tif
+            cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, band_tif] 
             logging.debug('Converting {} to {}'.format(f, band_tif))
             subprocess.call(cmd)
+            # add the new subdataset tif files to the list of tifs generated from this netcdf file
             band_tifs.append(band_tif)
-        merged_tif = '{}.tif'.format(os.path.splitext(f)[0])  # naming tiffs
+        # generate a name to save the tif file that will be produced by merging all the sub tifs from this netcdf   
+        merged_tif = '{}.tif'.format(os.path.splitext(f)[0])  
+        # merge all the sub tifs from this netcdf to create an overall tif representing all variables
         merge_cmd = ['gdal_merge.py', '-seperate'] + band_tifs + ['-o', merged_tif]
         subprocess.call(merge_cmd)
+        # add the new tif files to the list of tifs
         tifs.append(merged_tif)
     return tifs
 
 def list_available_files(url, ext=''):
+    '''
+    Fetch a list of filenames from source url by year
+    INPUT   url: url for fire weather data in the format "base url/{year}" (string)
```

Comment:
Making this description more general because this function could be used on other datasets besides fire weather
```suggestion
    INPUT   url: url for data source where we want to check for download links (string)
```